### DISCUSSION TITLE
Welcome to dperf Discussions!

### DISCUSSION THREAD

**Discussion by pengjianzhang** [baidu/dperf#403] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zrgtester** [comment]
请问为什么我使用payload_file参数像打纯tcp流量而非http流量时，通过抓包看到仅能看到三次握手就不再发包了。而使用payload_size参数就是正常的。配置如下

client conf：
mode                  client
tx_burst              32
launch_num            10
cpu                   1
socket_mem            2500
protocol              tcp
#payload_size          128
payload_file          /root/tcp_load1.txt
duration              36h
cps                   1
cc                    1
keepalive             50ms
port                  0000:05:00.0   172.16.1.2   172.16.1.1
client                20.1.1.1        100
server                192.168.1.1    1
listen                13000           15

server conf:
mode             server
tx_burst         32
launch_num       10
cpu              6
socket_mem       2500
protocol         tcp
duration         36h
payload_size    200
#payload_file     tcp_load.txt
keepalive        50ms
port                  0000:05:00.0   172.16.1.2   172.16.1.1
client                20.1.1.1        100
server                192.168.1.1    1
listen                13000           15

**Reply by zrgtester** [reply]
看了下代码，定位是在http_set_payload 函数中一个bug导致的。@pengjianzhang
 if (cfg->server) {
            strcpy(http_rsp, payload);
        } else {
            strcpy(http_rsp, payload);
        }
在else中应该拷贝到http_req中。修改后重新编译后可用。

**Reply by pengjianzhang** [reply]
请新开一个Discussion讨论，不是在这里跟贴，谢谢

**Reply by pengjianzhang** [reply]
已经修复了，请测一下最新代码
